### PR TITLE
Add release automation script.

### PR DESCRIPTION
**Pull request by bsneed**
- Added release.sh
- Updated releasing.md
- Added safety text in Version.swift to deter by-hand modification.

### CODE REVIEW

File: Sources/Segment/Version.swift [hunk 1]
```diff
@@ -7,5 +7,12 @@
 
 // Referred to by Analytics.swift
 
+// DO NOT MODIFY THIS FILE BY HAND!!
+// DO NOT MODIFY THIS FILE BY HAND!!
+// DO NOT MODIFY THIS FILE BY HAND!!
+// DO NOT MODIFY THIS FILE BY HAND!!
```
Comment (prayansh) [inline Sources/Segment/Version.swift:13]: 😂

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# check that we're on the `main` branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [ branch != 'main' ] 
+then
+	echo "The 'main' branch must be the current branch out to make a release."
+	echo "You are currently on: $branch"
+	exit 1
+fi
+
+versionFile="./sources/Segment/version.swift"
```
Comment (prayansh) [inline release.sh]: not sure if `V` needs to be uppercase

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+vercomp () {
+	if [[ $1 == $2 ]]
+	then
+		return 0
```
Comment (prayansh) [inline release.sh:6]: if the new version is the same as old version, we should error out, right?

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	echo "New version must be greater than previous version ($version)."
+	exit 1
+fi
+
+# get the commits since the last release...
+# note: we do this here so the "Version x.x.x" commit doesn't show up in logs.
+changelog=git log --pretty=format:"- (%an) %s" $(git describe --tags --abbrev=0 @^)..@
+tempFile=$(mktemp)
+#write changelog to temp file.
+echo $changelog >> $tempFile
```
Comment (prayansh) [inline release.sh:94]: whats the reason for a temp file?

**Review by prayansh — APPROVED** [review]
added some comments, but looks good

File: release.sh (hunk 3 above)
Comment (bsneed) [reply release.sh:6]: yup

File: release.sh (hunk 2 above)
Comment (bsneed) [reply release.sh]: case insensitive file system; but yeah it should be.

File: release.sh (hunk 4 above)
Comment (bsneed) [reply release.sh:94]: I write the changelog out there, and then gh release will read it in from there.

**Pull request merged**